// angular2-highcharts/__ivy_ngcc__/dist/ChartXAxisComponent.js
"use strict";
var ɵngcc0 = require('@angular/core');
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var ChartXAxisComponent = (function () {
    function ChartXAxisComponent() {
        this.afterBreaks = new core_1.EventEmitter();
        this.afterSetExtremes = new core_1.EventEmitter();
        this.pointBreak = new core_1.EventEmitter();
        this.pointInBreak = new core_1.EventEmitter();
        this.setExtremes = new core_1.EventEmitter();
    }
ChartXAxisComponent.ɵfac = function ChartXAxisComponent_Factory(t) { return new (t || ChartXAxisComponent)(); };
ChartXAxisComponent.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: ChartXAxisComponent, selectors: [["xAxis"]], outputs: { afterBreaks: "afterBreaks", afterSetExtremes: "afterSetExtremes", pointBreak: "pointBreak", pointInBreak: "pointInBreak", setExtremes: "setExtremes" } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ChartXAxisComponent, [{
        type: core_1.Directive,
        args: [{
                selector: 'xAxis'
            }]
    }], function () { return []; }, { afterBreaks: [{
            type: core_1.Output
        }], afterSetExtremes: [{
            type: core_1.Output
        }], pointBreak: [{
            type: core_1.Output
        }], pointInBreak: [{
            type: core_1.Output
        }], setExtremes: [{
            type: core_1.Output
        }] }); })();
    return ChartXAxisComponent;
}());
__decorate([
    core_1.Output(),
    __metadata("design:type", Object)
], ChartXAxisComponent.prototype, "afterBreaks", void 0);
__decorate([
    core_1.Output(),
    __metadata("design:type", Object)
], ChartXAxisComponent.prototype, "afterSetExtremes", void 0);
__decorate([
    core_1.Output(),
    __metadata("design:type", Object)
], ChartXAxisComponent.prototype, "pointBreak", void 0);
__decorate([
    core_1.Output(),
    __metadata("design:type", Object)
], ChartXAxisComponent.prototype, "pointInBreak", void 0);
__decorate([
    core_1.Output(),
    __metadata("design:type", Object)
], ChartXAxisComponent.prototype, "setExtremes", void 0);
exports.ChartXAxisComponent = ChartXAxisComponent;
